### PR TITLE
Moved the kafka service to the correct endpoints

### PR DESCRIPTION
**Pull request by ryan-a-git**

### CODE REVIEW

File: src/main/java/com/mercuryCyclists/Sales/service/InStoreSaleService.java [hunk 1]
```diff
@@ -166,16 +162,29 @@ public InStoreSale registerBackorder(InStoreSale inStoreSale, Long storeId){
...
+        ResponseEntity<String> productResponse = null;
+        try {
+            productResponse = restTemplate.postForEntity(POSTBACKORDER, msg, String.class);
+        } catch (Exception e) {
+            throw new IllegalStateException(e);
+        }
+
+        if(productResponse == null) {
+            throw new IllegalStateException("Product response was empty, please check that procurement service is running and Kafka is running");
+        }
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/service/InStoreSaleService.java]: put this in the generic sale service?

File: src/main/java/com/mercuryCyclists/Sales/service/OnlineSaleService.java [hunk 2]
```diff
@@ -25,13 +22,13 @@ public class OnlineSaleService {
...
-    private final KafkaTemplate<String, String> kafkaTemplate;
+    private static final String POSTBACKORDER = "http://localhost:8081/api/v1/product/backorder";
+    private static final RestTemplate restTemplate = new RestTemplate();
 
     @Autowired
-    public OnlineSaleService(OnlineSaleRepository onlineSaleRepository, SaleService saleService, KafkaTemplate<String, String> kafkaTemplate) {
+    public OnlineSaleService(OnlineSaleRepository onlineSaleRepository, SaleService saleService) {
         this.onlineSaleRepository = onlineSaleRepository;
         this.saleService = saleService;
-        this.kafkaTemplate = kafkaTemplate;
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/service/OnlineSaleService.java:37]: could you remove all the kafka related stuff in this PR as well? Thanks

**Review by pedrooste — DISMISSED** [review]
Small comments

File: src/main/java/com/mercuryCyclists/Sales/service/OnlineSaleService.java (hunk 2 above)
Comment (pedrooste) [reply src/main/java/com/mercuryCyclists/Sales/service/OnlineSaleService.java:37]: scrap tjat

**Review by Jack-Is-2048 — APPROVED** [review]

**Pull request merged**